File: Techniques/Two Pointers/twoSum2.go
package main

func twoSum2(nums []int, target int) []int {
	// array is one-indexed
	l, r := 0, len(nums)-1

	for r > l {
		s := nums[l] + nums[r]

		if s == target {
			return []int{l + 1, r + 1}
		}

		if s < target {
			l++
		} else {
			r--
		}
	}
	return []int{}
}

File: Techniques/Two Pointers/twoSum2.py
class Solution:
    def twoSum(self, arr: list[int], target: int) -> list[int]:
        # arr is one-indexed
        l, r = 0, len(arr) - 1

        while r > l:
            s = arr[l] + arr[r]

            if s == target:
                return [l + 1, r + 1]
            elif s < target:
                l += 1
            else:
                r -= 1
        return []